### PR TITLE
Update IgBlast to 1.20.0

### PR DESCRIPTION
**Pull request by bmcconeghy**
Bumping the version of IgBlast from 1.19.0 -> 1.20.0. Also dropping the build dep on `libxml2`.
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by bmcconeghy** [comment]
@tdsmith Here is the PR that will hopefully lead to unblocking us.

@joltex @stefanhannie @dcollins15 @ekieflabc @rgoya

**Comment by bmcconeghy** [comment]
@BiocondaBot please add label

**Comment by tdsmith** [comment]
As Brian hints in the first comment, our true ulterior motive is to get a conda package of igblast that's built against libxml 2.10.

We believe libxml 2.10 should be pulled in because conda-forge switched its pin to 2.10 in October, and we understand that bioconda uses the same build configuration: https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml

Brian noticed that this build is still using 2.9: https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=27232&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=7df82132-b284-504b-53d6-7d3e63519572&l=120

I think this is probably because a dependency is still pulling in a libxml 2.9 dependency; I think a likely candidate is ncbi-vdb's dependency on perl-xml-libxml, which hasn't been updated since last May. I think we may need to fire a PR to rebuild that (and its reverse dependencies) first before this bump will solve the problem we have.

This may be worth taking anyway in the interim since it delivers a new igblast :)

**Comment by bmcconeghy** [comment]
@BiocondaBot please fetch artifacts

**Comment by bmcconeghy** [comment]
@BiocondaBot please merge

**Comment by bmcconeghy** [comment]
@BiocondaBot  please request a review for this PR

**Comment by bmcconeghy** [comment]
> So, along with the update, you want to pin `libxml2 >=2.10`?

Ideally, yes

**Comment by bmcconeghy** [comment]
@BiocondaBot please fetch artifacts

**Comment by dpryan79** [comment]
We're currently pinning to conda-forge's pinnings in version 2022.08.25.15.20.42 (there's a conda-forge-pinning package on conda that's used to version this), so presumably before they updated to 2.10. I don't think we have imminent plans of updating that before the next R release, but I guess we could if there's a strong reason to switch to 2.10.

**Comment by tdsmith** [comment]
Ah, thank you, that's clarifying. I think https://github.com/bioconda/bioconda-recipes/blob/f9d68b5ec68a39b9ee56a962d49f20f89346c7e4/recipes/bioconda-utils/meta.yaml#L41 is the source of truth for the conda-forge-pinning pin, right?

Indeed, the libxml2 2.10 pin landed in October: https://github.com/conda-forge/conda-forge-pinning-feedstock/commit/0f31157c22e7aa78359a68a2c729a96a39ca39e4

@dpryan79 When you say next R release, do you mean e.g. 4.3.0?

We don't care about libxml2 per se; it just has a big reverse dependency tree, we're trying to migrate from python 3.9 to 3.10, and we have some pins on package versions that _do_ have (py3.9, xml2.9) builds but do _not_ have (py3.10, xml2.9) builds in conda-forge. I think this is substantially our problem, not yours.

I _think_ pinning libxml2 in only this formula would not completely solve our problem because we also need to be able to install igblast's dependencies, so we would also need to obtain libxml2 2.10 builds of those packages, notably perl-xml-libxml. I suppose we could submit that to conda-forge; some other perl-xml packages are present there.

I did notice that _this_ build did not seem to actually link libxml2; configure says it can't find libxml2, and the built artifacts don't link libxml2, but the build nonetheless completed:
https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=27628&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=7df82132-b284-504b-53d6-7d3e63519572&l=1190

I don't know enough about igblast to understand whether this is disabling useful functionality; @bmcconeghy, do you know? If igblast doesn't actually have a direct dependency on libxml2, I think that also helps.

**Comment by bmcconeghy** [comment]
> Ah, thank you, that's clarifying. I think
> 
> https://github.com/bioconda/bioconda-recipes/blob/f9d68b5ec68a39b9ee56a962d49f20f89346c7e4/recipes/bioconda-utils/meta.yaml#L41
> is the source of truth for the conda-forge-pinning pin, right?
> 
> Indeed, the libxml2 2.10 pin landed in October: [conda-forge/conda-forge-pinning-feedstock@0f31157](https://github.com/conda-forge/conda-forge-pinning-feedstock/commit/0f31157c22e7aa78359a68a2c729a96a39ca39e4)
> 
> @dpryan79 When you say next R release, do you mean e.g. 4.3.0?
> 
> We don't care about libxml2 per se; it just has a big reverse dependency tree, we're trying to migrate from python 3.9 to 3.10, and we have some pins on package versions that _do_ have (py3.9, xml2.9) builds but do _not_ have (py3.10, xml2.9) builds in conda-forge. I think this is substantially our problem, not yours.
> 
> I _think_ pinning libxml2 in only this formula would not completely solve our problem because we also need to be able to install igblast's dependencies, so we would also need to obtain libxml2 2.10 builds of those packages, notably perl-xml-libxml. I suppose we could submit that to conda-forge; some other perl-xml packages are present there.
> 
> I did notice that _this_ build did not seem to actually link libxml2; configure says it can't find libxml2, and the built artifacts don't link libxml2, but the build nonetheless completed: https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=27628&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=7df82132-b284-504b-53d6-7d3e63519572&l=1190
> 
> I don't know enough about igblast to understand whether this is disabling useful functionality; @bmcconeghy, do you know? If igblast doesn't actually have a direct dependency on libxml2, I think that also helps.

Unfortunately, I'm unsure how the dependency on libxml2 affects igblast. Judging by the fact that it wasn't linked, I gather it must be a transient dependency?

**Comment by tdsmith** [comment]
Combing through the igblast tarball, I can't find a single reference to libxml outside the build system, so I suspect this is just vestigial. I can't find a place where perl-xml-libxml would be used, either.

**Comment by dpryan79** [comment]
Worth removing then.

File: recipes/igblast/meta.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ requirements:
     - zlib
     - bzip2
     - libidn11
-    - libxml2
+    - libxml2 >=2.10
```
Comment (bmcconeghy) [inline recipes/igblast/meta.yaml:42]: Dropping this as it seems vestigial
```suggestion
```

**Comment by bmcconeghy** [comment]
@BiocondaBot please fetch artifacts

**Comment by bmcconeghy** [comment]
@BiocondaBot please request a review for this PR. Would we be able to merge this in?

**Comment by dpryan79** [comment]
@bmcconeghy Yes, merged!

**Pull request merged**